ArcGISRuntimeSDKQt_CppSamples/Scenes/DisplayScenesInTabletopAR/DisplayScenesInTabletopAR.cpp

```diff
@@ -0,0 +1,193 @@
+// [WriteFile Name=DisplayScenesInTabletopAR, Category=Scenes]
+// [Legal]
+// Copyright 2019 Esri.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// [Legal]
+
+#include "DisplayScenesInTabletopAR.h"
+
+#include "ArcGISTiledElevationSource.h"
+#include "Scene.h"
+#include "SceneQuickView.h"
+#include "ArcGISSceneLayer.h"
+#include "Error.h"
+#include "MobileScenePackage.h"
+
+#include <QDir>
+#include <QtCore/qglobal.h>
+
+#ifdef Q_OS_IOS
+#include <QStandardPaths>
+#endif // Q_OS_IOS
+
+using namespace Esri::ArcGISRuntime;
+
+// helper method to get cross platform data path
+namespace
+{
+QString defaultDataPath()
+{
+  QString dataPath;
+
+#ifdef Q_OS_ANDROID
+  dataPath = "/sdcard";
+#elif defined Q_OS_IOS
+  dataPath = QStandardPaths::writableLocation(QStandardPaths::DocumentsLocation);
+#else
+  dataPath = QDir::homePath();
+#endif
+
+  return dataPath;
+}
+} // namespace
+
+using namespace Esri::ArcGISRuntime;
+
+using namespace Esri::ArcGISRuntime::Toolkit;
+
+ArcGISArView* DisplayScenesInTabletopAR::arcGISArView() const
+{
+  return m_arcGISArView;
+}
+
+void DisplayScenesInTabletopAR::setArcGISArView(ArcGISArView* arcGISArView)
+{
+  if (!arcGISArView || arcGISArView == m_arcGISArView)
+    return;
+
+  m_arcGISArView = arcGISArView;
+
+  // full scale mode
```

Comment:
Couple of quetions about this:

- this isn't full scale AR, it is a tabletop. Is this block needed?
- Is it required that a default constructed Camera is passed in, and a nullptr for location data source is passed in? Seems like initialization code that shouldn't need to be in the client app

Reply:
You are right. They are not needed.